lib/matplotlib/cbook/__init__.py

```diff
@@ -2766,3 +2766,42 @@ def _topmost_artist(
     in reverse order.
     """
     return _cached_max(reversed(artists))
+
+
+def _define_aliases(local_d, alias_d):
+    """Define property aliases.
+
+    Use in a class definition as ::
+
+        cbook._define_aliases(locals(), {
+            "property": ["alias", ...], ...
+        })
+
+    For each property, if the corresponding ``get_property`` is defined in the
+    class so far, an alias named ``get_alias`` will be defined; the same will
+    be done for setters.  If neither the getter nor the setter exists, an
+    exception will be raised.
+
+    The alias map is stored as the ``_alias_map`` attribute on the class and
+    can be used by `~.normalize_kwargs`.
+    """
+
+    def make_alias(name):  # Enfore a closure over *name*.
+        def method(self, *args, **kwargs):
+            return getattr(self, name)(*args, **kwargs)
+        return method
+
+    for prop, aliases in alias_d.items():
+        exists = False
+        for prefix in ["get_", "set_"]:
+            if prefix + prop in local_d:
+                exists = True
+                for alias in aliases:
+                    method = make_alias(prefix + prop)
+                    method.__name__ = str(prefix + alias)  # Py2 compat.
+                    method.__doc__ = "alias for `{}`".format(prefix + prop)
```

Comment:
I suppose setting `__signature__` through `functools.wraps` also does the wrong thing for overriden properties...

Reply:
I'm mostly too lazy to add a dep on funcsigs for this PR (I actually do add such a dependency for the pyplot PR, so that'd also mean merge conflicts and sadness).